zipkin/src/main/java/zipkin/internal/DependencyLinker.java

```diff
@@ -104,8 +109,11 @@ public DependencyLinker putTrace(Iterator<DependencyLinkSpan> spans) {
         if (logger.isLoggable(FINE)) {
           logger.fine("processing ancestor " + ancestor.value());
         }
-        if (ancestor.value().kind == DependencyLinkSpan.Kind.SERVER) {
-          parent = ancestor.value().service;
+        DependencyLinkSpan ancestorLink = ancestor.value();
```

Comment:
similarly here, if we use a sentinel instead of null, I think the code is more self-describing..

```
if (ancestor == Node.MISSING_ROOT) {
  // explain
} else ... existing code
```